lib/recurly/element/element.js

```diff
@@ -0,0 +1,389 @@
+    }
+
+    parent.appendChild(container);
+    bus.add(this.window);
+    this.once(this.messageName('ready'), () => this.emit('attach', this));
+
+    return this;
+  }
+
+  /**
+   * Removes the element from the DOM
+   *
+   * @public
+   */
+  remove () {
+    const { attached, bus, id, container } = this;
+    debug('remove', id);
+    if (!attached) return this;
+    const parent = container.parentElement;
+    if (!parent) return this;
+    bus.remove(this.window);
+    parent.removeChild(container);
+    // Prevent dangling ready listeners from triggering the 'attach' event
+    this.off(this.messageName('ready'));
+    this.emit('remove', this);
+    return this;
+  }
+
+  /**
+   * Destroys the element and disables its functionality
+   *
+   * @public
+   */
+  destroy () {
+    this.remove();
+    this.off();
+    this.bus.remove(this);
+    this.elements.remove(this);
+    delete this.container;
+    delete this.iframe;
+    delete this.window;
+    return this;
+  }
+
+  /**
+   * Configures the element, enforcing the config whitelist
+   *
+   * @public
+   */
+  configure (options = {}) {
+    const { _config } = this;
+    const { OPTIONS } = this.constructor;
+    const newConfig = deepAssign({}, _config, pick(options, OPTIONS));
+    if (JSON.stringify(_config) === JSON.stringify(newConfig)) return this;
+    this._config = newConfig;
+    this.update();
+    return this;
+  }
+
+  /**
+   * Places focus on this element
+   */
+  focus () {
+    this.bus.send(this.messageName('focus!'));
+  }
+
+  /**
+   * Updates element DOM properties to latest values
+   *
+   * @private
+   */
+  update () {
+    const { bus, classList, config, id, iframe } = this;
+    const tabIndex = parseInt(config.tabIndex, 10) || 0;
+    debug('update', id);
+    this.container.className = classList;
+    iframe.setAttribute('tabindex', tabIndex);
+    bus.send(this.messageName('configure!'), config);
+  }
+
+  /**
+   * Builds a standard message name string in the format expected from
+   * a frame message
+   *
+   * @private
+   * @param {string} name message name
+   * @return {String} fully qualified message name
+   */
+  messageName (name) {
+    return `element:${this.id}:${name}`;
+  }
+
+  /**
+   * Provides an Array of tabbable HTMLElements in the document body,
+   * excluding (Recurly) Element frames
+   *
+   * @private
+   * @return {Array} of HTMLElements
+   */
+  tabbableItems () {
+    const { FRAME_NAME_PREFIX } = this.constructor;
+    return tabbable(window.document.body).filter(el => el.name.indexOf(FRAME_NAME_PREFIX) !== 0);
+  }
+
+  // Event handlers
+
+  /**
+   * Updates state to reflect changes signaled by the Element iframe
+   *
+   * @private
+   * @param  {Object} body message body, containing new state
+   */
+  onStateChange (body) {
+    debug('state change', this.id, body);
+    let newState = { ...body };
+    delete newState.type;
+    if (JSON.stringify(this.state) === JSON.stringify(newState)) return;
```

Comment:
Is there any value in abstracting out these deep equal object comparisons into a util function? Seems like it'd be an unnecessary abstraction, but figured I may as well throw it out there.

Reply:
Mm since there's only two in here, it's okay as-is for now. But I'd totally welcome making a util for the op as a separate PR 💥